### PR TITLE
dotnet-sdk_3: 3.1.420 -> 3.1.422

### PR DESCRIPTION
**Pull request by mdarocha**
###### Description of changes

Small bugfix release. See [release notes](https://github.com/dotnet/core/blob/main/release-notes/3.1/3.1.27/3.1.27.md)

Follow up to https://github.com/NixOS/nixpkgs/pull/181512, and requires it merged first.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mdarocha** [comment]
Packages to update dependencies:

- [ ] python-language-server w
- [ ] asabibackend

**Comment by newAM** [comment]
Result of `nixpkgs-review pr 181520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>alttpr-opentracker</li>
    <li>azure-functions-core-tools</li>
    <li>baget</li>
    <li>dotnet-aspnetcore_3 (dotnetCorePackages.aspnetcore_3_1)</li>
    <li>dotnet-runtime_3 (dotnetCorePackages.runtime_3_1)</li>
    <li>dotnet-sdk_3 (dotnetCorePackages.sdk_3_1)</li>
    <li>inklecate</li>
    <li>python-language-server</li>
    <li>vscode-extensions.ms-vsliveshare.vsliveshare</li>
    <li>wasabibackend</li>
  </ul>
</details>

File: pkgs/development/compilers/dotnet/update.sh [hunk 1]
```diff
@@ -92,21 +92,28 @@ aspnetcore_packages () {
...
-      "Microsoft.AspNetCore.App.Runtime.osx-arm64" \
       "Microsoft.AspNetCore.App.Runtime.osx-x64" \
       "Microsoft.AspNetCore.App.Runtime.win-arm" \
       "Microsoft.AspNetCore.App.Runtime.win-arm64" \
       "Microsoft.AspNetCore.App.Runtime.win-x64" \
       "Microsoft.AspNetCore.App.Runtime.win-x86" \
     )
 
+    # Packages that only apply to .NET 5 and up
+    if [[ ! ( "$version" =~ ^[1-3]\. ) ]]; then
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/dotnet/update.sh:108]: Can we make this comparison less brittle?

File: pkgs/development/compilers/dotnet/update.sh [hunk 2]
```diff
@@ -223,6 +208,34 @@ sdk_packages () {
       "runtime.win-x86.Microsoft.NETCore.DotNetHostResolver" \
     )
 
+    # Packages that only apply to .NET 5 and up
+    if [[ ! ( "$version" =~ ^[1-3]\. ) ]]; then
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/dotnet/update.sh:212]: Same here

File: pkgs/development/compilers/dotnet/update.sh (hunk 1 above)
Comment (mdarocha) [reply pkgs/development/compilers/dotnet/update.sh:108]: Hmm, I don't think this comparision is any more brittle than other ones in this script. I also don't really have an idea how to improve it, so suggestions welcome :)

Comment (IvarWithoutBones) [reply pkgs/development/compilers/dotnet/update.sh:108]: Maybe we can `nix eval` something like `lib.versionOlder`?

Comment (SuperSandro2000) [reply pkgs/development/compilers/dotnet/update.sh:108]: That sounds like a better idea and less likely to break.

Comment (IvarWithoutBones) [reply pkgs/development/compilers/dotnet/update.sh:108]: For reference, that should also happen on line 240. Should probably create a `versionOlder` function in bash that does the `nix eval` to reduce code repetition.

Comment (mdarocha) [reply pkgs/development/compilers/dotnet/update.sh]: Ok, I updated the checks to use `versionOlder`.

**Review by IvarWithoutBones — APPROVED** [review]
LGTM :+1: 

Result of `nixpkgs-review pr 181520` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>alttpr-opentracker</li>
    <li>azure-functions-core-tools</li>
    <li>baget</li>
    <li>dotnet-aspnetcore_3 (dotnetCorePackages.aspnetcore_3_1)</li>
    <li>dotnet-runtime_3 (dotnetCorePackages.runtime_3_1)</li>
    <li>dotnet-sdk_3 (dotnetCorePackages.sdk_3_1)</li>
    <li>inklecate</li>
    <li>python-language-server</li>
    <li>vscode-extensions.ms-vsliveshare.vsliveshare</li>
    <li>wasabibackend</li>
  </ul>
</details>

**Pull request merged**